src/Microsoft.Health.Dicom.Blob/Features/ExternalStore/ExternalStoreHealthExpiryHttpPipelinePolicy.cs

```diff
@@ -24,11 +25,23 @@ internal class ExternalStoreHealthExpiryHttpPipelinePolicy : HttpPipelinePolicy
     public ExternalStoreHealthExpiryHttpPipelinePolicy(ExternalBlobDataStoreConfiguration externalStoreOptions)
     {
         _externalStoreOptions = EnsureArg.IsNotNull(externalStoreOptions, nameof(externalStoreOptions));
-        EnsureArg.IsNotNull(externalStoreOptions.BlobContainerUri, nameof(externalStoreOptions.BlobContainerUri));
         EnsureArg.IsNotNull(externalStoreOptions.StorageDirectory, nameof(externalStoreOptions.StorageDirectory));
         EnsureArg.IsNotNull(externalStoreOptions.HealthCheckFilePath, nameof(externalStoreOptions.HealthCheckFilePath));
 
-        UriBuilder uriBuilder = new UriBuilder(_externalStoreOptions.BlobContainerUri);
+        Uri blobUri;
+
+        if (_externalStoreOptions.BlobContainerUri != null)
+        {
+            blobUri = _externalStoreOptions.BlobContainerUri;
+        }
+        else
+        {
+            // For local testing with Azurite
+            BlobContainerClient blobContainerClient = new BlobContainerClient(_externalStoreOptions.ConnectionString, _externalStoreOptions.ContainerName);
```

Comment:
Should this be more guarded with env than the uri?

Reply:
This is how its done in the `ExternalBlobClient` https://github.com/microsoft/dicom-server/blob/7d750be725e34aebcb78663a670bb9d393e62606/src/Microsoft.Health.Dicom.Blob/Features/ExternalStore/ExternalBlobClient.cs#L25

We could guard by the current environment, although technically environment doesn't matter here. If we for some reason wanted to switch to connecting to the blob store via ConnectionString instead of BlobContainerUri, the url generated in the `else` statement would still be accurate.